contracts/contracts/strategies/crosschain/CrossChainMasterStrategy.sol

```diff
@@ -171,8 +178,35 @@ contract CrossChainMasterStrategy is
         uint256 feeExecuted,
         bytes memory payload
     ) internal override {
-        // Expecting a BALANCE_CHECK_MESSAGE
+        uint64 _nonce = lastTransferNonce;
+
+        // Should be expecting an acknowledgement
+        require(!isNonceProcessed(_nonce), "Nonce already processed");
+        // Only a withdrawal can send tokens to Master strategy
+        require(
+            transferTypeByNonce[_nonce] == TransferType.Withdrawal,
+            "Expecting withdrawal"
+        );
+
+        // Confirm receipt of tokens from Withdraw command
+        _markNonceAsProcessed(_nonce);
+
+        // Now relay to the regular flow
+        // NOTE: Calling _onMessageReceived would mean that we are bypassing a
+        // few checks that the regular flow does (like sourceDomainID check
+        // and sender check in `handleReceiveFinalizedMessage`). However,
+        // CCTPMessageRelayer relays the message first (which will go through
+        // all the checks) and not update balance and then finally calls this
+        // `_onTokenReceived` which will update the balance.
+        // So, if any of the checks fail during the first no-balance-update flow,
+        // this won't happen either, since the tx would revert.
         _onMessageReceived(payload);
+
+        // Send any tokens in the contract to the Vault
+        uint256 usdcBalance = IERC20(baseToken).balanceOf(address(this));
+        // Should always have enough tokens
+        require(usdcBalance >= tokenAmount, "Insufficient balance");
+        IERC20(baseToken).safeTransfer(vaultAddress, tokenAmount);
```

Comment:
I think we might want to transfer everything we have in this contract: 
```
IERC20(baseToken).safeTransfer(vaultAddress, usdcBalance);
```
This way there will never be dust left